### ISSUE TITLE
CSV logger bad output on warnings

### ISSUE THREAD

**Issue by wummel** [wummel/linkchecker#320]
Converted from [SourceForge issue 3438553](http://sourceforge.net/support/tracker.php?aid=3438553), submitted by sychan168

The CSV logger outputs malforms, truncated lines when there are warnings. The problem seems to be in
.../site-packages/linkcheck/logger/csvlog.py
csvlog.log_url() has the following lines for warningstrins and infostrings:

```
    if self.has_part("warningstring"):
        row.append(self.linesep.join(x[1] for x in url_data.warnings))
    if self.has_part("infostring"):
        row.append(self.linesep.join(url_data.info))
```

   The string arrays are joined by line separators. When the csv.writer writes a row, the row gets truncated after the first carriage return, so you only see the first warning message, and there is no close quote. This breaks parsers that are "correct".

```
I've modified it as follows and it seems to function well enough:


    if self.has_part("warningstring"):
        row.append(",".join(x[1] for x in url_data.warnings))
    if self.has_part("infostring"):
        row.append(",".join(url_data.info))
```
